### ISSUE TITLE
[Regression]: Browser fetcher collects Chrome revisions for Firefox

### ISSUE THREAD

**Issue by timvandermeij** [puppeteer/puppeteer#9602]
### Bug expectation

During the upgrade of Puppeteer from version 19.0.0 to version 19.1.0 in Mozilla's PDF.js project (https://github.com/mozilla/pdf.js/issues/15865) we ran into the issue that our two browsers (Firefox and Chrome) didn't start correctly anymore. I already filed #9461 before that fortunately got fixed now, but that uncovered another very similar issue in the browser fetcher class this time, and most likely also caused by #9140 that caused the previous issue. I found out that this is a regression and will provide a reduced test case. Note that this still happens with the latest version 19.6.2, so it's not resolved by the fix for #9461.

Steps to reproduce the problem:

1. Create a file called `repro.js` with the contents of the reduced test case provided below.
2. Install Puppeteer 19.0.0 and let it install Chrome and Firefox.
3. Run `node repro.js` and notice that Firefox and Chrome revisions are reported correctly:

```
Found chrome revision 1045629
Found firefox revision 111.0a1
```

4. Install Puppeteer 19.1.0.
5. Run `node repro.js` and notice that the Firefox revision is now incorrect (equal to the Chrome revision):

```
Found chrome revision 1045629
Found firefox revision 1045629
```

This is bad because we delete all revisions except for the current ones from the cache folder at each run, so incorrectly reported revisions cause this process to fail and the disk to fill up over time.

Since the root cause of this bug, mixing up Firefox and Chrome state internally in Puppeteer, is very similar to what happened in #9461 I sadly suspect that https://github.com/puppeteer/puppeteer/commit/ec201744f077987b288e3dff52c0906fe700f6fb#diff-545f1eb6b4d2825410b4be2eb6f6baa65ab44841c7ef5ff444bb01a5ece66c2aR198 is also the culprit here.

/cc @jrandolf as the author of the commit who might have an idea on how to best resolve this issue.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
const puppeteer = require("puppeteer");

async function getRevisions(browserName) {
  const browserFetcher = puppeteer.createBrowserFetcher({
    product: browserName,
  });
  const localRevisions = await browserFetcher.localRevisions();
  for (const localRevision of localRevisions) {
    console.log(`Found ${browserName} revision ${localRevision}`);
  }
}

getRevisions("chrome");
getRevisions("firefox");
```

### Error string

no error

### Puppeteer version

19.6.2

### Node version

19.5.0

### Package manager

npm

### Package manager version

8.19.2

### Operating system

Linux

**Comment by timvandermeij** [comment]
Moreover, if you want please feel free to use this reduced test case and the one from #9461 in your test suite. Given these two regressions there is probably some missing coverage for handling multiple browsers, so hopefully these two test cases can help to increase that coverage.

**Comment by jrandolf** [comment]
We will triage this with multi-browser support. As stated in the previous issue, Puppeteer isn't really built for this use case, however since BiDi is in development, this is in our radar. Just stay tooned.

**Comment by timvandermeij** [comment]
Puppeteer 20 seems to work now if we start the browsers in series, which is fine. We used to start them in parallel, but that still causes version number confusion in Puppeteer's node class, however I can imagine that's just not how it's supposed to work.

However, I have opened https://github.com/puppeteer/puppeteer/issues/10174 because the browsers package sadly does not fully have the same capabilities yet and we used `BrowserFetcher.{localRevisions,remove}` to clean up old browsers.
